### DISCUSSION TITLE
Allow user to manually scrub through autoplay/replays even if legacy progress meter is in use

### DISCUSSION THREAD

**Discussion by marioman63** [ppy/osu#28001] [Ideas]
Noticed this due to playing maps that use their own skins, but if the skin does not have the modern progress bar (the screenwide bar with timer and progress percentage) and instead uses the circle in the top right (a la stable), you cannot scrub an autoplay attempt. You can still use the controls on the right, but without the modern laser progress bar, manual scrubbing is not possible.

![image](https://github.com/ppy/osu/assets/76608775/f6258418-b9c0-46ff-98a5-00976aa5652b)

above image uses this map https://osu.ppy.sh/beatmapsets/475469#osu/1018342 with its included skin enabled as an example. You can see how it uses the legacy progress meter instead of the laser one, which is the one that supports scrubbing.

Example of map without a map skin, instead using a user skin that has the modern bar setup. notice the handle on the timeline that lets me scrub: 
![image](https://github.com/ppy/osu/assets/76608775/b3dd3e59-c752-4a36-b311-5a49d778feb7)

**Comment by DGCK81LNN** [comment]
As I prefer the new Argon skin and don't want to see any legacy skin elements anymore, I generally always Ignore Beatmap Skin on lazer. Ignoring beatmap colors is another toggle so it doesn't bother me with default combo colors like in stable.

I wonder how this can be implemented though. Perhaps display a progress bar when you place cursor on the bottom of the screen. Or maybe make it an option to replace the pie chart with a progress bar?

**Reply by peppy** [reply]
you can add a subtle argon progress bar from the skin editor.

you can also just use keyboard shortcuts.

**Reply by marioman63** [reply]
For context on my experience: I have many, many songs (Safe to say 4 digit numbers at least) that have built in skins. Having to add an Argon progress element for each is not really feasible (I do like the Argon meter and manually added it to my user skin, but I am talking specifically about map skins). If there are keyboard shortcuts that recreate the manual scrubbing feature (NOT the playback controls at the top right), I am unaware of them. I prefer the precision and granularity of the manual scrub compared to the playback controls as it lets me quickly search songs for specific parts, or precisely control the speed I view things at in a way not achievable by the playback controls.

**Comment by cl8n** [comment]
think the best way to fix this would be for replays to provide an argon progress bar on hover at the bottom of the screen, if your skin doesn't already have a bar. this feels natural to me bc many video players work this way

**Reply by peppy** [reply]
If we go with that, it'd probably be easiest to remove the interactivity from the skinned progress bars and move it to a new global component 🤷
